### PR TITLE
Update Average.java

### PR DESCRIPTION
**Pull request by bayramtturgutt**
- Made the constructor throw an UnsupportedOperationException to prevent instantiation, making it explicit that this is a utility class.
- Added a private validateInput method to handle validation, reducing code duplication and improving readability.
- Consistent exception messages and handling for both methods.
- Improved comments to be more descriptive and follow JavaDoc conventions.
- Enhanced code readability and maintained consistent formatting.

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.
- [x] All new code is formatted with `clang-format -i --style=file path/to/your/file.java`

### CODE REVIEW

File: src/main/java/com/thealgorithms/maths/Average.java [hunk 1]
```diff
@@ -25,15 +31,15 @@ public static double average(double[] numbers) {
...
-     * @param numbers the array contains element and the sum does not excess long
-     *                value limit
-     * @return average value
+     * @param numbers an array of {@code int} values
+     * @return the average of the given numbers
+     * @throws IllegalArgumentException if the input array is {@code null} or empty
      */
     public static int average(int[] numbers) {
         if (numbers == null || numbers.length == 0) {
-            throw new IllegalArgumentException("Numbers array cannot be empty or null");
```
Comment (BamaCharanChhandogi) [inline src/main/java/com/thealgorithms/maths/Average.java:36]: You can keep this line. Do not need to change

File: src/main/java/com/thealgorithms/maths/Average.java [hunk 2]
```diff
@@ -25,15 +31,15 @@ public static double average(double[] numbers) {
...
+     * Computes the average of an {@code int} array.
      *
-     * @param numbers the array contains element and the sum does not excess long
-     *                value limit
-     * @return average value
+     * @param numbers an array of {@code int} values
+     * @return the average of the given numbers
+     * @throws IllegalArgumentException if the input array is {@code null} or empty
      */
     public static int average(int[] numbers) {
```
Comment (BamaCharanChhandogi) [inline src/main/java/com/thealgorithms/maths/Average.java:40]:
```suggestion
    public static double average(int[] numbers) {
```
Change the data type

**Review by BamaCharanChhandogi — CHANGES_REQUESTED** [review]
Or this also

return **(double)** sum / numbers.length;

File: src/main/java/com/thealgorithms/maths/Average.java (hunk 1 above)
Comment (bayramtturgutt) [reply src/main/java/com/thealgorithms/maths/Average.java:36]: Got it! I'll keep it as is. 👍

File: src/main/java/com/thealgorithms/maths/Average.java (hunk 2 above)
Comment (bayramtturgutt) [reply src/main/java/com/thealgorithms/maths/Average.java]: Roger, applied

**Review by BamaCharanChhandogi — APPROVED** [review]
Looks good, thanks.

**Pull request merged**